power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -615,14 +603,21 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
     TapPositionOptimizerImpl(Calculator calculator, StateUpdater updater, OptimizerStrategy strategy)
         : calculate_{std::move(calculator)}, update_{std::move(updater)}, strategy_{strategy} {}
 
-    auto optimize(State const& state, CalculationMethod method) -> ResultType final {
+    auto optimize(State const& state, CalculationMethod method) -> MathOutput<ResultType> final {
+
         auto const order = regulator_mapping<TransformerTypes...>(state, TransformerRanker{}(state));
 
         auto const cache = this->cache_states(order);
-        auto optimal_output = optimize(state, order, method);
+        auto solver_output = optimize(state, order, method);
         update_state(cache);
 
-        return optimal_output;
+        TransformerTapPositionResult transformer_tap_positions;
+        (get_transformer_tap_positions<TransformerTypes, typename State::ComponentContainer>(state,
+                                                                                             transformer_tap_positions),
+         ...);
```

Comment:
`update_state(cache)` should happen after this (or, equivalently, this output needs to happen in the `optimize` implementation)

Reply:
Yes, I think you are correct in this. Could you apply the change?